ianalyzer_readers/readers/rdf.py

```diff
@@ -0,0 +1,45 @@
+from typing import Iterable
+
+from rdflib import BNode, Graph
+
+from .core import Reader, Document, Source
+
+
+class RDFReader(Reader):
+    '''
+    A base class for Readers of Resource Description Framework files.
+    These could be in Turtle, JSON-LD, RDFXML or other formats,
+    see [rdflib parsers](https://rdflib.readthedocs.io/en/stable/plugin_parsers.html)
+    '''
+
+    def source2dicts(self, source: Source) -> Iterable[Document]:
+        '''
+        Given a RDF source file, returns an iterable of extracted documents.
+
+        Parameters:
+            source: the source file to extract. This is a string of the file path
```

Comment:
Normally, a `Source` type can can also be a tuple with a filename and metadata, or a bytestream (though support for that one is spotty). It's not necessarily a problem to support limited source types in a reader, but supporting a tuple with metadata is useful, and should be easy enough to add.